### PR TITLE
Added new configuration parameters for Canary Upgrade functionality in Kiali Overview page.

### PR DESCRIPTION
**Pull request by hhovsepy**
Core PR https://github.com/kiali/kiali/pull/4122
UI PR https://github.com/kiali/kiali-ui/pull/2184

### CODE REVIEW

File: roles/default/kiali-deploy/defaults/main.yml [hunk 1]
```diff
@@ -205,6 +205,7 @@ kiali_defaults:
 
   kiali_feature_flags:
     istio_injection_action: true
+    istio_upgrade_action: false
```
Comment (jmazzitelli) [inline roles/default/kiali-deploy/defaults/main.yml:208]: Even though the other two settings are not defined by default (I am assuming this to be true), you still should put them in here for documentation purposes, for developers to see.

So up around lines [154-160](https://github.com/kiali/kiali-operator/blob/v1.36.0/roles/default/kiali-deploy/defaults/main.yml#L154-L160), add these two settings, but comment them out (just like we do for things like [label_selector as seen here](https://github.com/kiali/kiali-operator/blob/v1.36.0/roles/default/kiali-deploy/defaults/main.yml#L28)):

```
      envoy_admin_local_port: 15000
      #istio_current_revision:
      istio_identity_domain: "svc.cluster.local"
      istio_injection_annotation: "sidecar.istio.io/inject"
      istio_sidecar_annotation: "sidecar.istio.io/status"
      istio_sidecar_injector_config_map_name: "istio-sidecar-injector"
      #istio_upgrade_revision:
      istiod_deployment_name: "istiod"
      url_service_version: ""
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
Some style/formatting things to change.

Also, you might want to consider renaming those 2 settings. You have:

* istio_current_revision
* istio_upgrade_revision

To make it more obvious to the user that these two are related to each other (i.e. if you set one, you need to set the other), I would consider naming these:

* istio_revision_current
* istio_revision_upgrade

In fact, you might even want to make them into a single dict value to, again, make it obvious that you need to set both (i.e. setting one but not the other would not be valid).

```
istio_revision:
  current:
  upgrade:
```

Not a requirement, just a thought for you to consider.

File: deploy/kiali/kiali_cr.yaml [hunk 2]
```diff
@@ -539,10 +539,13 @@ spec:
 #                  namespace: <the namespace where your ingress is deployed>
 # config_map_name: The name of the istio control plane config map. It defaults to `istio`.
 # envoy_admin_local_port: The port which kiali will open to fetch envoy config data information.
+# istio_canary_revision: These vales are used in Canary upgrade/downgrade functionality when istio_upgrade_action is true.
```
Comment (jmazzitelli) [inline deploy/kiali/kiali_cr.yaml:542]: Everything looks good to me - only thing is a misspelling of the word `values`:

```suggestion
# istio_canary_revision: These values are used in Canary upgrade/downgrade functionality when istio_upgrade_action is true.
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]

**Comment by hhovsepy** [comment]
thanks @jmazzitelli , all requested changes are done now, the third suggestion seemed to be the best

**Review by jmazzitelli — APPROVED** [review]
I haven't run this, but the code looks good.

**Review by lucasponce — DISMISSED** [review]
We are discussing if it's good to consolidate this feature.
@jmazzitelli if the CR code looks ok, feature it's pretty simple and disable by default, but it could be good to ship it on 1.37 to let users provide feedback.

**Review by jmazzitelli — APPROVED** [review]
lgtm

**Comment by hhovsepy** [comment]
@lucasponce I did rebase and appovals are removed, can you please merge it?

**Review by lucasponce — APPROVED** [review]

**Pull request merged**